### DISCUSSION TITLE
Which triplet of euler angles is chosen among all equivalents for cubic symmetry?

### DISCUSSION THREAD

**Discussion by EmileHazemann** [mtex-toolbox/mtex#1527] [Ask Anything]
Hello,

I have a question regarding a ctf file for a nickel single-crystal sample obtained with Brüker software, but couldn't find the answer online, maybe you could answer me.
I obtained several orientation maps for nickel samples oriented along <001>, <110> and <111> directions, that I post-process on MTex. I checked the orientation of these samples by XRD with pole figures. My objective is to compare Euler angles obtained by EBSD and by XRD measurements.
On MTex, I choose the convention 'convertspatial2eulerreferenceframe', plotx2east and plotzoutofplane to get the same map orientation and euler angles as in Brüker software. When I check all the equivalent orientations with cubic symmetry, I get 48 different triplets of euler angles. How the triplet of the single crystal displayed in the ctf file and MTex is calculated among these 48 possibilities?
In XRD measurements, as I know my sample reference frame, I can choose which crystallographic direction is close to which sample direction ([001] with Z direction, [100] with X direction and [010] with Y direction) and calculate Euler angles accordingly. Then I want to find a systematic relationship (the same rotation matrix for all samples) to get euler angles obtained by EBSD equivalent to the ones calculated by XRD. As it seems to work for samples oriented along <001> direction, I struggle to find this relationship for <110> and <111> oriented samples. It would help to know if the triplet of euler angles given in ctf file and imported in MTex is random or not.

Thanks in advance,

Emile

**Comment by pilchak2** [comment]
Emile,  I suggest comparing all of the symmetric orientations against your XRD one and then take the minimum of that set.   However, this seems like a straight geometry problem and a  transformation could be developed if you have a good understanding of the sample reference frame in each case. Can you sketch out the sample along with  the X-Y-Z axes of each instrument and then go through the series of rotations that would align a grain at (0,0,0), i.e. with [100] || X and [010] || Y, etc,  with one another?

**Reply by EmileHazemann** [reply]
Thank you very much for your answer! Indeed, it seems like a straight geometry problem.
I sketched out the different reference frames for a sample oriented along [001] in the attached file. (Xl, Yl, Zl) represents the sample reference frame, (Xdrx, Ydrx, Zdrx) the XRD reference frame according to the position of my sample during measuring, (Xm, Ym, Zm) the Euler angle reference frame in MTex. For XRD measurement, I use the entire sample, and for EBSD measurement I use a longitudinal cut as presented in the attached file. There is several unclarities for me about EBSD measurement, especially for (Xl, Yl, Zl) in a CTF view (southern part of the sketch, in which (Xs, Ys) represents the image reference frame from CTF and (Xm, Ym, Zm) the Euler angle reference frame in MTex, chosen to match both image orientation and euler angles of the ctf file). I represented the reference fram that seems natural to me, considering that (Xs, Ys, Zs) in a ctf file are aligned to the measurement reference frame in the SEM. But as I understand, I'm not sure it is the case, it could change between different set ups of the SEM.
It is rather unexpected that the Euler angles given by MTex (and CTF file) match my XRD measurement with just an additional 180° rotation around Zm, without using the rotation matrix to go from (Xm, Ym, Zm) to (Xdrx, Ydrx, Zdrx).

Thanks in advance.

![sketch_rotations](https://user-images.githubusercontent.com/111073296/189097477-0ef1353e-c0fa-4901-848f-84ead969cb4e.png)

**Comment by ralfHielscher** [comment]
Hi Emile,

may you clarify your question a little bit. From what I understood you input is:
1. orientations from EBSD
2. orientations from XRD (I did not know, that this is possible -  or is this 4d XRD?)

Now you are a bit unsure about the alignment of the specimen reference system for these two set of orientations. In other words if you plot both sets of orientations into pole figures you are searching for an rotation (which would then be the change of reference frame) that make both sets of pole figures to agree.

Is this correct?

Ralf.

**Reply by EmileHazemann** [reply]
Hi Ralf,

Thank you for your answer.
My inputs are:
1. Orientations from EBSD
2.  Poles figures (generally {200}, {111} and {220}) - so no, it is classic XRD, with PANalytical Empyrean system

I am sure about the alignment of the specimen reference frame with the XRD reference frame. However, I have doubts regarding specimen reference frame in the SEM (refering to http://dx.doi.org/10.1016/j.matchar.2016.04.008).

Yes, exactly, I adjust manually Euler angles to match the poles of my pole figure.

Thanks in advance,

Emile
